uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/LeaveRequestTest.php

```diff
@@ -4361,4 +4362,64 @@ public function testCalculateBalanceChangeForTwoDaysLeaveRequestWhenStartAndEndD
 
     $this->assertEquals($expectedResultsBreakdown, $result);
   }
+
+  public function testNonMandatoryFieldsAreResetWhenAbsenceTypeChangedToOneWithADifferentCalculationUnit() {
+    $calculationUnits = array_flip(AbsenceType::buildOptions('calculation_unit', 'validate'));
+    $dayTypes = array_flip(LeaveRequest::buildOptions('from_date_type', 'validate'));
+
+    $absenceTypeInDays = AbsenceTypeFabricator::fabricate([
+      'calculation_unit' => $calculationUnits['days']
+    ]);
+
+    $absenceTypeInHours = AbsenceTypeFabricator::fabricate([
+      'calculation_unit' => $calculationUnits['hours']
+    ]);
+
+    $leaveRequestParams = [
+      'contact_id' => 1,
+      'type_id' => $absenceTypeInDays->id,
+      'from_date' => CRM_Utils_Date::processDate('2016-01-01'),
+      'to_date' => CRM_Utils_Date::processDate('2016-01-01'),
+      //The types are required when absence type is in days
+      'from_date_type' => $dayTypes['all_day'],
+      'to_date_type' => $dayTypes['all_day']
+    ];
+
+    $leaveRequest = LeaveRequestFabricator::fabricateWithoutValidation($leaveRequestParams);
+
+    // Change absence type to one in hours
+    $leaveRequestParams = array_merge($leaveRequestParams, [
+      'id' => $leaveRequest->id,
+      'type_id' => $absenceTypeInHours->id,
+      // These two are required for Leave in Hours
+      'to_date_amount' => 7.5,
+      'from_date_amount' => 0,
+    ]);
+
+    $leaveRequest = LeaveRequestFabricator::fabricateWithoutValidation($leaveRequestParams);
```

Comment:
This is a bit misleading since it looks like it's creating a new `$leaveRequest`. I know it's the same with the `create` method of the API, so not sure what to suggest here. What do you think?

Reply:
I changed it to `create`. As you said, it's pretty much the same thing, but it was the best I could do.